spec/graphql/queries/search_query_spec.rb

```diff
@@ -0,0 +1,143 @@
+
+    it 'returns the correct number of entries' do
+      expect(search_result['entries'].length).to eq(10)
+      expect(search_result['count']['all']).to eq(10)
+      expect(search_result['count']['organizationalUnits']).to eq(5)
+      expect(search_result['count']['repositories']).to eq(5)
+    end
+
+    it 'returns the repositories' do
+      repositories = search_result['entries'].select do |e|
+        e['entry']['__typename'] == 'Repository'
+      end
+      expect(repositories.length).to eq(5)
+    end
+
+    it 'returns the organizational units' do
+      organizational_units = search_result['entries'].select do |e|
+        %w(User Organization).include?(e['entry']['__typename'])
+      end
+      expect(organizational_units.length).to eq(5)
+    end
+  end
+
+  context 'category: all' do
+    let(:variables) { {'query' => '', 'category' => 'all'} }
+
+    it 'returns the correct number of entries' do
+      expect(search_result['entries'].length).to eq(10)
+      expect(search_result['count']['all']).to eq(10)
+      expect(search_result['count']['organizationalUnits']).to eq(5)
+      expect(search_result['count']['repositories']).to eq(5)
+    end
+
+    it 'returns the repositories' do
+      repositories = search_result['entries'].select do |e|
+        e['entry']['__typename'] == 'Repository'
+      end
+      expect(repositories.length).to eq(5)
+    end
+
+    it 'returns the organizational units' do
+      organizational_units = search_result['entries'].select do |e|
+        %w(User Organization).include?(e['entry']['__typename'])
+      end
+      expect(organizational_units.length).to eq(5)
+    end
+  end
+
+  context 'category: repositories' do
+    let(:variables) { {'query' => '', 'category' => 'repositories'} }
+
+    it 'returns the correct number of entries' do
+      expect(search_result['entries'].length).to eq(5)
+      expect(search_result['count']['all']).to eq(10)
+      expect(search_result['count']['organizationalUnits']).to eq(5)
+      expect(search_result['count']['repositories']).to eq(5)
+    end
+
+    it 'returns the repositories' do
+      repositories = search_result['entries'].select do |e|
+        e['entry']['__typename'] == 'Repository'
+      end
+      expect(repositories.length).to eq(5)
+    end
+  end
+
+  context 'category: organizationalUnits' do
+    let(:variables) { {'query' => '', 'category' => 'organizationalUnits'} }
+
+    it 'returns the correct number of entries' do
+      expect(search_result['entries'].length).to eq(5)
+      expect(search_result['count']['all']).to eq(10)
+      expect(search_result['count']['organizationalUnits']).to eq(5)
+      expect(search_result['count']['repositories']).to eq(5)
+    end
+
+    it 'returns the organizational units' do
+      organizational_units = search_result['entries'].select do |e|
+        %w(User Organization).include?(e['entry']['__typename'])
+      end
+      expect(organizational_units.length).to eq(5)
+    end
+  end
+end
```

Comment:
This is only a comment, not a change request:

The most part of the specs is duplicated code. You could write a shared example block that uses let-defined variables that you check on, as in:
```ruby
    it 'returns the correct number of entries' do
      expect(search_result['entries'].length).to eq(expected_num_entries)
      expect(search_result['count']['all']).to eq(expected_count_all)
      expect(search_result['count']['organizationalUnits']).to eq(expected_count_organizational_units)
      expect(search_result['count']['repositories']).to eq(expected_count_repositories)
    end
```
This would make the test clearer.